src/valid/expression.rs

```diff
@@ -168,6 +167,9 @@ impl super::Validator {
                         return Err(ExpressionError::InvalidIndexType(index));
                     }
                 }
+                if !is_pointer && function.expressions[index].is_dynamic_index(module) {
```

Comment:
what if it's a vector? can still be dynamically indexed

Reply:
Arg, thanks!